homeassistant/components/overkiz/water_heater_entities/atlantic_dhw.py

```diff
@@ -0,0 +1,182 @@
+        | WaterHeaterEntityFeature.ON_OFF
+    )
+    _attr_operation_list = [*OVERKIZ_TO_OPERATION_MODE.keys()]
+
+    @property
+    def max_temp(self) -> float:
+        """Return the maximum temperature."""
+        return cast(
+            float,
+            self.executor.select_state(
+                OverkizState.CORE_MAXIMAL_TEMPERATURE_MANUAL_MODE
+            ),
+        )
+
+    @property
+    def min_temp(self) -> float:
+        """Return the minimum temperature."""
+        return cast(
+            float,
+            self.executor.select_state(
+                OverkizState.CORE_MINIMAL_TEMPERATURE_MANUAL_MODE
+            ),
+        )
+
+    @property
+    def current_temperature(self) -> float:
+        """Return the current temperature."""
+        return cast(
+            float,
+            self.executor.select_state(
+                OverkizState.MODBUSLINK_MIDDLE_WATER_TEMPERATURE
+            ),
+        )
+
+    @property
+    def target_temperature(self) -> float:
+        """Return the temperature corresponding to the PRESET."""
+        return cast(
+            float,
+            self.executor.select_state(OverkizState.CORE_WATER_TARGET_TEMPERATURE),
+        )
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set new temperature."""
+        temperature = kwargs[ATTR_TEMPERATURE]
+        await self.executor.async_execute_command(
+            "setTargetDHWTemperature", temperature
+        )
+
+    @property
+    def is_boost_mode_on(self) -> bool:
+        """Return true if boost mode is on."""
+        return (
+            self.executor.select_state(OverkizState.MODBUSLINK_DHW_BOOST_MODE)
+            == OverkizCommandParam.ON
+        )
+
+    @property
+    def is_eco_mode_on(self) -> bool:
+        """Return true if eco mode is on."""
+        return self.executor.select_state(OverkizState.MODBUSLINK_DHW_MODE) in (
+            OverkizCommandParam.MANUAL_ECO_ACTIVE,
+            OverkizCommandParam.AUTO_MODE,
+        )
+
+    @property
+    def is_away_mode_on(self) -> bool:
+        """Return true if away mode is on."""
+        return (
+            self.executor.select_state(OverkizState.MODBUSLINK_DHW_ABSENCE_MODE)
+            == OverkizCommandParam.ON
+        )
+
+    @property
+    def dhw_mode(self) -> str:
+        """Return DWH mode."""
+        return cast(str, self.executor.select_state(OverkizState.MODBUSLINK_DHW_MODE))
+
+    @property
+    def current_operation(self) -> str:
+        """Return current operation."""
+        if self.is_boost_mode_on:
+            output = STATE_PERFORMANCE
+        elif self.is_eco_mode_on:
+            output = STATE_ECO
+        elif self.is_away_mode_on:
+            output = STATE_OFF
+        elif (mode := self.dhw_mode) == OverkizCommandParam.MANUAL_ECO_INACTIVE:
+            output = OverkizCommandParam.MANUAL
+        else:
+            output = mode
+
+        return output
```

Comment:
Sorry, I saw no harm in returning a custom mode that is not supported by exactly this device. This device reports ManualEcoInactive for Manual, and ManualEcoActive for Eco, but there can also be Eco+ mode that can report something else, that I couldn't find out, or any other mode, that is not described in this particular device, but shares controllable_name with it. Any other mode not supported here would report STATE_OFF instead of the actual mode name. I will do as you say, of course.